web/pdf_print_service.js

```diff
@@ -190,24 +190,27 @@ class PDFPrintService {
   useRenderedPage() {
     this.throwIfInactive();
     const img = document.createElement("img");
-    const scratchCanvas = this.scratchCanvas;
-    if ("toBlob" in scratchCanvas) {
-      scratchCanvas.toBlob(function (blob) {
-        img.src = URL.createObjectURL(blob);
-      });
-    } else {
-      img.src = scratchCanvas.toDataURL();
-    }
+    this.scratchCanvas.toBlob(blob => {
+      img.src = URL.createObjectURL(blob);
+    });
 
     const wrapper = document.createElement("div");
     wrapper.className = "printedPage";
     wrapper.append(img);
     this.printContainer.append(wrapper);
 
-    return new Promise(function (resolve, reject) {
-      img.onload = resolve;
-      img.onerror = reject;
-    });
+    const { promise, resolve, reject } = Promise.withResolvers();
+    img.onload = resolve;
+    img.onerror = reject;
+
+    promise
+      .catch(() => {
+        // Avoid "Uncaught promise" messages in the console.
+      })
+      .then(() => {
```

Comment:
Shouldn't we revoke the URL in a `finally`clause instead ?

Reply:
Then we get "Uncaught promise" messages in the console mentioned in the comment just above, which the `catch` -> `then` structure helps avoids. (This pattern is already used elsewhere in the code-base in similar cases.)

Reply:
I'm not sure to understand your point... mine was just about using a `finally` instead of `then` (and keep the `catch`) in order to make sure the URL is revoked whatever the image stuff ended.

Reply:
The end result will be exactly the same though, but with a couple more characters of code!?